packages/taquito/test/contract/rpc-contract-provider.spec.ts

```diff
@@ -1790,6 +1792,103 @@ describe('RpcContractProvider test', () => {
     });
   });
 
+  describe('updateConsensusKey', () => {
+    it('should produce an updateConsensusKey operation', async (done) => {
+      mockRpcClient.getManagerKey.mockReturnValue('test_pub_key_hash');
+      mockEstimate.reveal.mockResolvedValue(undefined);
+      const estimate = new Estimate(1230000, 93, 142, 250);
+      mockEstimate.updateConsensusKey.mockResolvedValue(estimate);
+      const result = await rpcContractProvider.updateConsensusKey({
+        pk: 'edpkti5K5JbdLpp2dCqiTLoLQqs5wqzeVhfHVnNhsSCuoU8zdHYoY7',
+      });
+
+      expect(result.raw).toEqual({
+        opbytes: 'test',
+        opOb: {
+          branch: 'test',
+          contents: [
+            {
+              kind: 'update_consensus_key',
+              source: 'test_pub_key_hash',
+              fee: '475',
```

Comment:
suggestion for fee, gas_limit and storage_limit properties, you could take the value from estimate :
```
fee: estimate.suggestedFeeMutez.toString(),
gas_limit: estimate.gasLimit.toString(),
storage_limit: estimate.storageLimit.toString(),
```